### PR TITLE
fix: Improve the URI template params validation

### PR DESCRIPTION
**Pull request by w-vi**
Covers:

- action only
- resource with params
- resource with params under action

### CODE REVIEW

File: src/ParametersParser.h [hunk 1]
```diff
@@ -157,6 +157,64 @@ namespace snowcrash {
...
+    /**
+     * \brief Check Parameter validity in URI template
+     */
+
+    // It must either be in path "{param" or at the begining
+    // "?param" or in the list ",param". And any of the params
+    // must end either with "}" or ",".
+
+    //FIXME: The implementation is very naive and can be sped up.
+    static bool isValidUriTemplateParam(const std::string &uriTemplate, const std::string &param) {
```
Comment (pksunkara) [inline src/ParametersParser.h]: Wait, why is this written again? The previous PR was merged into master.

Comment (pksunkara) [inline src/ParametersParser.h]: Did you forget to remove these 2 functions from `ResourceParser`?

Comment (w-vi) [inline src/ParametersParser.h]: Yes

Comment (w-vi) [inline src/ParametersParser.h]: Addressed

**Pull request merged**